KMS VGA666 overlay

6by9 opened the pull request:
Adds an overlay for the VGA666 under full KMS.

If you don't have DDC connected through a level shifter then you'll need a ```video=VGA-1....``` entry in cmdline.txt, or add it via xrandr.

6by9 committed:
dt: Add option for dpi without DE and PCLK (for VGA666)

VGA666 doesn't use the DE or PCLK signals, therefore there is
no point in claiming their use. It's also then possible to
use GPIOs 0&1 for DDC to read the EDID from the display.

Signed-off-by: Dave Stevenson <dave.stevenson@raspberrypi.com>

6by9 committed:
dtoverlays: Add an overlay for the VGA666 when used with vc4-kms-v3d

Includes optional use of GPIOs 0&1 / BSC0 for DDC to read the EDID
from the display.

Signed-off-by: Dave Stevenson <dave.stevenson@raspberrypi.com>

6by9 committed:
defconfigs: Add DRM_DISPLAY_CONNECTOR and DRM_SIMPLE_BRIDGE for VGA666

VGA666 uses "vga-connector" from DRM_DISPLAY_CONNECTOR, and
"dumb-vga-dac" from DRM_SIMPLE_BRIDGE to connect up, so add them
to the defconfigs.

Signed-off-by: Dave Stevenson <dave.stevenson@raspberrypi.com>

6by9 committed:
drm/vc4: Change the default DPI format to being 18bpp, not 24.

DPI hasn't really been used up until now, so the default has
been meaningless.
In theory we should be able to pass the desired format for the
adjacent bridge chip through, but framework seems to be missing
for that.

As the main device to use DPI is the VGA666 or Adafruit Kippah,
both of which use RGB666, change the default to being RGB666 instead
of RGB888.

Signed-off-by: Dave Stevenson <dave.stevenson@raspberrypi.com>

pelwell commented on arch/arm/boot/dts/overlays/README:
```diff
@@ -3120,6 +3120,15 @@ Params: cma-512                 CMA is 512MB
+Info:   Enable the VGA666 (resistor ladder ADC) for the vc4-kms-v3d driver.
+        Requires vc4-kms-v3d to be loaded.
+Load:   dtoverlay=vc4-kms-vga666,<param>
+Params: ddc                    Enables GPIOs 0&1 as the I2C to read the EDID
```
The indentation on the parameter description is off-by-one, but otherwise this looks OK.

6by9 replied:
Fixed.

pelwell commented:
Although these fragments aren't listed in application order, it doesn't matter because the intra-first rule is sufficient provided you don't have fragments that target fragments that target fragments.

6by9 commented:
Updated.
I've also added a note under vga666 (which just sets the GPIOs to alt2) that it is not for use with vc4-kms-v3d as you'll get a clash over the GPIOs.
